src/ArticleTheme.vala

```diff
@@ -0,0 +1,109 @@
+//	This file is part of FeedReader.
+//
+//	FeedReader is free software: you can redistribute it and/or modify
+//	it under the terms of the GNU General Public License as published by
+//	the Free Software Foundation, either version 3 of the License, or
+//	(at your option) any later version.
+//
+//	FeedReader is distributed in the hope that it will be useful,
+//	but WITHOUT ANY WARRANTY; without even the implied warranty of
+//	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//	GNU General Public License for more details.
+//
+//	You should have received a copy of the GNU General Public License
+//	along with FeedReader.  If not, see <http://www.gnu.org/licenses/>.
+using Gee;
+
+public class  FeedReader.ArticleTheme {
+    private static ArrayList<HashMap<string, string>> ? themes = null;
```

Comment:
Consider making the ThemeInfo into a struct:

```
public struct ThemeInfo {
  string name;
  string author;
  bool is_corrupted;
  // etc.
}

private static ArrayList<ThemeInfo>? themes = null;
// etc.
```

Since we know the contents of the map, using a struct is more efficient and easier to work with.

You might also consider making this a map from theme name to theme info so we can more easily lookup themes:

```
private static HashMap<string, ThemeInfo>? = null;

public static HashMap<string, ThemeInfo> getThemes() { ... }

public static ThemeInfo? getTheme(string name)
{
  var themes = getThemes();
  if (!themes.has(name))
    return null;
  return themes.get(name);
}

public static bool exists(string name)
{
  return getTheme(name) != null;
}
```